contracts/main/price-feeders/DelayFathomOraclePriceFeed.sol

```diff
@@ -52,48 +50,88 @@ contract DelayFathomOraclePriceFeed is PausableUpgradeable, IFathomOraclePriceFe
     event LogSetPriceLife(address indexed _caller, uint256 _second);
     event LogSetTimeDelay(address indexed _caller, uint256 _second);
 
+    function readPrice() external view override returns (bytes32) {
+        return bytes32(delayedPrice);
+    }
+
+    function peekPrice() external override returns (bytes32, bool) {
+       return _peekPrice();
+    }
+
+    function isPriceOk() external view override returns (bool) {
+        return _isPriceOk();
+    }
+
+    function setAccessControlConfig(address _accessControlConfig) external onlyOwner {
+        IAccessControlConfig(_accessControlConfig).hasRole(IAccessControlConfig(_accessControlConfig).OWNER_ROLE(), msg.sender); // Sanity Check Call
+        accessControlConfig = IAccessControlConfig(_accessControlConfig);
+    }
+
     function setPriceLife(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
+        require(_second >= timeDelay && _second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-price-life");
         priceLife = _second;
         emit LogSetPriceLife(msg.sender, _second);
     }
 
     function setTimeDelay(uint256 _second) external onlyOwner {
-        require(_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
+        require(_second <= priceLife &&_second >= 5 minutes && _second <= 1 days, "FathomOraclePriceFeed/bad-delay-time");
+        _peekPrice();
         timeDelay = _second;
         emit LogSetTimeDelay(msg.sender, _second);
     }
 
+    function setToken0(address _token) external onlyOwner {
+        require(_token != address(0), "FathomOraclePriceFeed: ZERO_ADDRESS");
+        require(token1 != _token, "FathomOraclePriceFeed/same-token0-token1");
+
+        token0 = _token;
+    }
+
+    function setToken1(address _token) external onlyOwner {
+        require(_token != address(0), "FathomOraclePriceFeed: ZERO_ADDRESS");
+        require(token0 != _token, "FathomOraclePriceFeed/same-token0-token1");
+
+        token1 = _token;
+    }
+
+    function setOracle(address _oracle) external onlyOwner {
```

Comment:
2.3.5 fathomOracle , token0 and token1 cannot be changed in DelayFathomOraclePriceFeed - fixed by adding a setter, now we can set the new oracle in case we fail to pass correct parameters during the contracts deploy, and no need to redeploy price feed